contracts/apps/voting/VotingApp.sol

```diff
@@ -0,0 +1,190 @@
+    * @param _supportRequiredPct Percentage of voters that must support a voting for it to succeed (expressed as a 10^18 percetage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _minAcceptQuorumPct Percetage of total voting power that must support a voting  for it to succeed (expressed as a 10^18 percetage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _voteTime Seconds that a voting will be open for token holders to vote (unless it is impossible for the fate of the vote to change)
+    */
+    function initialize(MiniMeToken _token, uint256 _supportRequiredPct, uint256 _minAcceptQuorumPct, uint64 _voteTime) onlyInit {
+        initialized();
+
+        require(_supportRequiredPct <= PCT_BASE);
+        require(_supportRequiredPct >= _minAcceptQuorumPct);
+
+        token = _token;
+        supportRequiredPct = _supportRequiredPct;
+        minAcceptQuorumPct = _minAcceptQuorumPct;
+        voteTime = _voteTime;
+
+        votings.length += 1;
+    }
+
+    /**
+    * @notice Change minimum acceptance quorum to `_minAcceptQuorumPct`
+    * @param _minAcceptQuorumPct New acceptance quorum
+    */
+    function changeMinAcceptQuorumPct(uint256 _minAcceptQuorumPct) auth external {
+        require(supportRequiredPct >= _minAcceptQuorumPct);
+        minAcceptQuorumPct = _minAcceptQuorumPct;
+    }
+
+    /**
+    * @notice Create new voting to execute `_executionScript`
+    * @param _executionScript EVM script to be executed on approval
+    * @return votingId id for newly created vote
+    */
+    function newVoting(bytes _executionScript) auth external returns (uint256 votingId) {
+        votingId = votings.length++;
+        Voting storage voting = votings[votingId];
+        voting.executionScript = _executionScript;
+        voting.creator = msg.sender;
+        voting.startDate = uint64(now);
+        voting.open = true;
+        voting.snapshotBlock = getBlockNumber() - 1; // avoid double voting in this very block
+        voting.totalVoters = token.totalSupplyAt(voting.snapshotBlock);
+
+        StartVote(votingId);
+
+        if (canVote(votingId, msg.sender)) _vote(votingId, true, msg.sender);
+    }
+
+    /**
+    * @notice Vote `_supports` in vote with id `_votingId`
+    * @param _votingId Id for vote
+    * @param _supports Whether voter supports the voting
+    */
+    function vote(uint256 _votingId, bool _supports) external {
+        require(canVote(_votingId, msg.sender));
+        _vote(_votingId, _supports, msg.sender);
+    }
+
+    /**
+    * @notice Execute the result of vote `_votingId`
+    * @param _votingId Id for vote
+    */
+    function executeVote(uint256 _votingId) external {
+        require(canExecute(_votingId));
+        _executeVote(_votingId);
+    }
+
+    function canVote(uint256 _votingId, address _voter) constant returns (bool) {
+        Voting storage voting = votings[_votingId];
+
+        return voting.open &&
+               uint64(now) < voting.startDate + voteTime &&
```

Comment:
Could you add parentheses to make sure the order of operations is explicit?

Reply:
yep